### ISSUE TITLE
Does FlowER use Conditional Flow Matching (CFM) during inference?

### ISSUE THREAD

**Issue by SixuanMi** [FongMunHong/FlowER#1]
Hi,

I've been testing the `run_FlowER_large_newData.sh` code and noticed some behavior that makes me curious about whether Conditional Flow Matching (CFM) is used during inference.

Here's the setup:

1. I fixed the random seed of torch in `beam_predict.py`
```
import torch
seed = 100
torch.manual_seed(seed)
if torch.cuda.is_available():
    torch.cuda.manual_seed(seed)
    torch.cuda.manual_seed_all(seed)
```
2. I ran tests using the beam.txt input file as specified in the documentation

The results showed differences when:

1. Providing the full reaction (`O.Cl.[OH3+].CC(=O)CC(=O)C(F)(F)F.NNc1cccc(Br)c1>>Cc1cc(n(n1)-c1cccc(c1)Br)C(F)(F)F|Cc1cc(nn1-c1cc(ccc1)Br)C(F)(F)F`) in beam.txt, the results were `Beam Search Results 0: 2/2 - {'Cc1cc(C(F)(F)F)n(-c2cccc(Br)c2)n1', 'Cc1cc(C(F)(F)F)nn1-c1cccc(Br)c1'}`.  
2. Providing only reactants (`O.Cl.[OH3+].CC(=O)CC(=O)C(F)(F)F.NNc1cccc(Br)c1`) in beam.txt, the results were `Beam Search Results 0: 0/0 - set()`.  

From reviewing your paper, I observe that CFM is explicitly used during the training process, but its application during inference is not mentioned. This has left me uncertain about its role in the inference/beam search phase. Is it necessary to provide products in beam.txt for FlowER to perform inference correctly since CFM is used?

Thanks for your help!

**Comment by FongMunHong** [comment]
Hi, thanks for checking out our work!

As for the not seeing the results check out this [line](https://github.com/FongMunHong/FlowER/blob/master/beam_predict.py#L226) which specifies which products to check when you run beam search. If not specified, it will be empty, thus no products is shown. Read more in the README [file](https://github.com/FongMunHong/FlowER/tree/master?tab=readme-ov-file#flower-search-input).

As for how CFM is being used in inference, I would encourage you to read the method section for better detailed understanding. A brief, one line summary to 'use' CFM in inference, is through integrating an Ordinary Differential Equation (ODE), in which the equation is approximated by our FlowER model.

**Comment by SixuanMi** [comment]
Thanks!
